### DISCUSSION TITLE
Indexing

### DISCUSSION THREAD

**Discussion by dharik-arsath** [vaexio/vaex#1412] [Q&A] [answered]
How to drop row by its index

In my case I have few rows with ' ' empty strings I can able to find that by using df[df.colname == " "] but now I have to delete those rows by index. How can I do that?

One way I can think of is df.colname.str.replace(" ",Nan) but here how to set nan value in vaex?
what are all the ways to drop a row?

**Comment by JovanVeljanoski** [comment, marked as answer]
With vaex the data is immutable, so you don't delete it. 

You can filter out rows pretty easily though. You had the right idea:
```python
df[~(df.x == '')]
```
